### PR TITLE
Add Node-RED and Mosquitto (MQTT) IIAB playbooks for IoT

### PR DESCRIPTION
**Pull request by m-anish**
Mosquitto is a pub-sub broker that a large number of IOT devices and sensors use. 
Node-red provides easy to make dashboards with a drag and drop interface, and a user-facing dashboard. 

With this functionality, the IIAB can serve as a broker to various sensors and actuators that may be connected to the network of which IIAB is a part of or manages. 

This set of playbooks have been tested to work on Intel-NUC on ubuntu 16.04, and are fairly straightforward, so I assume they should work on other iiab supported platforms too.

### CODE REVIEW

File: roles/mosquitto/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+mosquitto_install: True
+mosquitto_enabled: False
+mosquitto_user: admin
```
Comment (holta) [inline roles/mosquitto/defaults/main.yml]: Please change 'admin' to 'Admin' to be consistent with other Internet-in-a-Box playbooks, Thanks!

File: roles/nodered/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+nodered_install: True
+nodered_enabled: False
+nodered_user: admin
```
Comment (holta) [inline roles/nodered/defaults/main.yml]: Please change 'admin' to 'Admin' to be consistent with other Internet-in-a-Box playbooks, Thanks!

**Comment by holta** [comment]
Open Request: we need someone to test this PR/functionality on IIAB 6.6/master on RPi 3 or RPi 3 B+

(Additionally, the resulting microSD should be tested within an RPi Zero W if available!)

**Comment by holta** [comment]
Looks great, thx @m-anish, I'll ask ppl to smoke-test on RPi 3 B+ and Ubuntu 18.04 LTS in coming hours/days so we can merge!

Those who test in RPi 3 (or 3 B+) please also insert the microSD into RPi Zero W (after IIAB is installed) as a general rule, to make sure it at least boots!

PS I've pre-emptively fixed the documentation for all 3 local_vars.yml's {MIN, MEDIUM, BIG} to match your PR:

http://wiki.laptop.org/go/IIAB/local_vars_min.yml
http://wiki.laptop.org/go/IIAB/local_vars.yml
http://wiki.laptop.org/go/IIAB/local_vars_big.yml

**Comment by holta** [comment]
On IIAB 6.6/master on Raspbian Lite on RPi 3 B+, "./runrole mosquitto" completed (in 86 seconds) but "./runrole nodered" failed as follows:

> TASK [nodered : Install nodejs-legacy] *****************************************
failed: [127.0.0.1] (item=[u'nodejs-legacy']) => {"cache_update_time": 1530063312, "cache_updated": false, "changed": false, "item": ["nodejs-legacy"], "msg": "'/usr/bin/apt-get -y -o \"Dpkg::Options::=--force-confdef\" -o \"Dpkg::Options::=--force-confold\"     install 'nodejs-legacy'' failed: E: Unable to correct problems, you have held broken packages.\n", "rc": 100, "stderr": "E: Unable to correct problems, you have held broken packages.\n", "stderr_lines": ["E: Unable to correct problems, you have held broken packages."], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nSome packages could not be installed. This may mean that you have\nrequested an impossible situation or if you are using the unstable\ndistribution that some required packages have not yet been created\nor been moved out of Incoming.\nThe following information may help to resolve the situation:\n\nThe following packages have unmet dependencies:\n nodejs-legacy : Depends: nodejs (>= 0.6.19~dfsg1-3~) but it is not going to be installed\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information...", "Some packages could not be installed. This may mean that you have", "requested an impossible situation or if you are using the unstable", "distribution that some required packages have not yet been created", "or been moved out of Incoming.", "The following information may help to resolve the situation:", "", "The following packages have unmet dependencies:", " nodejs-legacy : Depends: nodejs (>= 0.6.19~dfsg1-3~) but it is not going to be installed"]}

On IIAB 6.6/master on Ubuntu 18.04 LTS/Desktop, "./runrole mosquitto" completed (in 31 seconds) but "./runrole nodered" failed as follows:

> TASK [nodered : Install nodejs-legacy] *****************************************
failed: [127.0.0.1] (item=[u'nodejs-legacy']) => {"changed": false, "item": ["nodejs-legacy"], "msg": "No package matching 'nodejs-legacy' is available"}

**Comment by holta** [comment]
@m-anish if it helps you debug, Raspbian Lite on RPi 3 B+ shows: (BIG-sized local_vars.yml)
```
root@box:~# apt -a list nodejs
Listing... Done
nodejs/unknown,now 6.14.3-1nodesource1 armhf [installed]
nodejs/stable 4.8.2~dfsg-1 armhf
```
And Ubuntu 18.04 LTS/Desktop shows: (also BIG-sized local_vars.yml)
```
root@box:~# apt -a list nodejs
Listing... Done
nodejs/bionic,now 8.10.0~dfsg-2 amd64 [installed]
```
(@jvonau might have ideas?)

**Comment by holta** [comment]
@m-anish what does your PR here need to become part of IIAB 6.6/master as "final test" and then release are now approaching?

Getting the correct Node.js 8.x similar to Sugarizer 1.0 (#798) facing much the same?

Please test against the latest Raspbian (which upgraded from Node.js 6.x to 8.x yesterday after "apt update") and/or Ubuntu 18.04 so we can merge this.

**Comment by holta** [comment]
CLARIF: it's important we dump Node 4.x as it was EOL'd (end-of-life'd) on 2018-04-30.

Consequently both Ubuntu 18.04 and Raspbian made the (smart) decision to skip Node.js 6.x, and went directly from 4.x to 8.x, so it's time we get with the program :)

FWIW, according to https://en.wikipedia.org/wiki/Node.js#Releases and https://github.com/nodejs/Release:
- Node.js 6.x EOL's April 2019
- Node.js 8.x EOL's 2019-12-31 "currently scheduled"
- Node.js 10.x EOL's April 2021

**Comment by holta** [comment]
@m-anish with our [IIAB 6.6](https://github.com/iiab/iiab/wiki/IIAB-6.6-Release-Notes) release imminent in a few very short weeks/days (target July 31) can you finalize and test this PR so it can be mainline'd?

(Or perhaps better to defer to [IIAB 6.7](https://github.com/iiab/iiab/milestone/4) later in 2018?)

**Comment by holta** [comment]
@m-anish am moving this to our [IIAB 6.7](https://github.com/iiab/iiab/milestone/4) milestone as IIAB 6.6's Release Candidate will be published in not so many days...if not hours...keep in touch :-)

**Comment by holta** [comment]
> @m-anish can you fix this to work on Ubuntu 18.04 for IIAB 6.7's release in coming weeks?

@m-anish should we defer to the next quarterly release, expected to be IIAB 7.0 in March 2019?

**Comment by m-anish** [comment]
I created a new PR #1398 that is rebased on current master. 

It installs cleanly on a ubuntu 18.04 VM, in a minimal IIAB setup. 

Further tests on a RPI and Debian would help validate the fixes in the new commits.

**Pull request closed without merging**